src/main/java/dev/ebyrdeu/movie_jakarta/repository/MovieRepository.java

```diff
@@ -0,0 +1,54 @@
+package dev.ebyrdeu.movie_jakarta.repository;
+
+import dev.ebyrdeu.movie_jakarta.entity.Movie;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.TypedQuery;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+public class MovieRepository {
+    private EntityManager entityManager;
+
+    public MovieRepository(EntityManager entityManager){
+        this.entityManager = entityManager;
+    }
+
+    public Movie saveMovie(Movie movie) {
+        try {
+            entityManager.getTransaction().begin();
+            if (movie.getId() == null) {
+                entityManager.persist(movie);
+            }else{
+                movie = entityManager.merge(movie);
+            }
+            entityManager.getTransaction().commit();
+            return movie;
+        }catch (RuntimeException e) {
+            entityManager.getTransaction().rollback();
+            throw e;
+        }
+    }
```

Comment:
use Transactional annotation instead - less code, do same and even better job 
```suggestion

    @Transactional
    public Movie saveMovie(Movie movie) {
       entityManager.persist(movie);
       return movie;
    }
```